frontend/coprs_frontend/commands/failed_to_succeeded_stats.py

```diff
@@ -0,0 +1,154 @@
+"""
+Generate failed to succeeded stats
+"""
+
+import os
+from datetime import datetime
+
+import click
+import pygal
+from coprs import models
+
+
+@click.command()
+@click.option("--dest", "-D", required=True, help="Result directory")
+def failed_to_succeeded_stats(dest):
+    """
+    Generate failed to succeeded stats
+    """
+    print("Please wait, this will take at least 20 minutes.")
+    categories = {
+        "immediately": 0,
+        "seconds": 0,
+        "minutes": 0,
+        "hours": 0,
+        "days": 0,
+        "weeks": 0,
+    }
+    tuples = failed_to_succeeded_tuples()
+    for failed, succeeded in tuples:
+        delta = datetime.fromtimestamp(succeeded) \
+            - datetime.fromtimestamp(failed)
+        categories[delta_to_category(delta)] += 1
+
+    os.makedirs(dest, exist_ok=True)
+    path = os.path.join(dest, "failed-to-succeeded-stats.svg")
+    generate_graph(categories, path)
+    print("Created: {0}".format(path))
+
+
+def get_builds():
+    """
+    Return list of all builds
+    """
+    query = (
+        models.Build.query
+        .join(models.Package)
+        .join(models.Copr)
+        .order_by(models.Build.id)
+    )
+
+    # Only recent builds
+    # start = datetime(2022, 1, 1).timestamp()
+    # end = datetime(2023, 1, 1).timestamp()
+    # query = (query
+    #          .filter(models.Build.submitted_on >= start)
+    #          .filter(models.Build.submitted_on <= end))
+
+    # Packit user
+    # query = query.filter(models.Copr.user_id==5576)
```

Comment:
with this, are the builds filtered via the Copr project owner? (or via the submitter of the build?)

Reply:
afaik the `user_id` in Copr model is only project owner since this model represents the project as a whole, not the individual builds. To filter builds according to submitter:

```py
query = query.filter(models.Build.user_id=PACKIT_USER_ID)
```

Reply:
Exactly as @nikromen is saying.